File: Examples/Features/Picture/PictureDuplicate.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Picture
{
    public class PictureDuplicate : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];

            System.IO.Stream stream = this.GetResourceStream("logo.png");

            //Create a shape in worksheet
            GrapeCity.Documents.Excel.Drawing.IShape picture = worksheet.Shapes.AddPicture(stream, GrapeCity.Documents.Excel.Drawing.ImageType.PNG, 20, 20, 395, 60);

            //Duplicate picture
            GrapeCity.Documents.Excel.Drawing.IShape newPicture = picture.Duplicate();

        }

        public override bool IsNew
        {
            get
            {
                return true;
            }
        }
    }
}

File: Examples.VB/Features/Picture/PictureDuplicate.vb
Namespace Features.Picture
    Public Class PictureDuplicate
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)
            Dim stream As IO.Stream = GetResourceStream("logo.png")

            'Create a shape in worksheet
            Dim picture As IShape = worksheet.Shapes.AddPicture(stream, ImageType.PNG, 20, 20, 395, 60)

            'Duplicate picture
            Dim newPicture As IShape = picture.Duplicate()
        End Sub
        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
